Subject:  Varying on a second vscsi controller causes a kernel panic.
From: boutcher@us.ibm.com
References: 182124 - LTC 23946

Without this patch we register an interrupt with request_irq,
but then return a bad return code from the module probe. 

Signed-off-by: Olaf Hering <olh@suse.de>

---
 drivers/scsi/ibmvscsi/rpa_vscsi.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-2.6.16/drivers/scsi/ibmvscsi/rpa_vscsi.c
===================================================================
--- linux-2.6.16.orig/drivers/scsi/ibmvscsi/rpa_vscsi.c
+++ linux-2.6.16/drivers/scsi/ibmvscsi/rpa_vscsi.c
@@ -239,6 +239,7 @@ int ibmvscsi_init_crq_queue(struct crq_q
 	if (rc == 2) {
 		/* Adapter is good, but other end is not ready */
 		printk(KERN_WARNING "ibmvscsi: Partner adapter not ready\n");
+		retrc = 0;
 	} else if (rc != 0) {
 		printk(KERN_WARNING "ibmvscsi: Error %d opening adapter\n", rc);
 		goto reg_crq_failed;
